src/m365/aad/commands/app/app-list.ts

```diff
@@ -0,0 +1,44 @@
+import { Application } from '@microsoft/microsoft-graph-types';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { odata } from "../../../../utils/odata";
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+export interface CommandArgs {
+  options: GlobalOptions;
+}
+
+class AadAppListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.APP_LIST;
+  }
+
+  public get description(): string {
+    return 'Gets a list of Azure AD app registrations';
+  }
+
+  constructor() {
+    super();
+  }
+
+  public async commandAction(logger: Logger): Promise<void> {
+    try {
+      const appList = await this.getAppList();
+      logger.log(appList);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private getAppList(): Promise<Application[]> {
```

Comment:
I suggest that we just remove this method and move the logic to the `commandAction`.